x-pack/legacy/plugins/graph/public/components/graph_settings/advanced_settings_form.tsx

```diff
@@ -0,0 +1,214 @@
+        <EuiFieldNumber
+          fullWidth
+          min={1}
+          max={500000}
+          step={1}
+          value={advancedSettings.minDocCount}
+          onChange={getNumberUpdater('minDocCount')}
+        />
+      </EuiFormRow>
+
+      <EuiFormRow
+        fullWidth
+        helpText={
+          <>
+            {i18n.translate('xpack.graph.settings.advancedSettings.diversityFieldInputHelpText1', {
+              defaultMessage:
+                'To avoid document samples being dominated by a single voice, pick the field that helps identify the source of bias.',
+            })}{' '}
+            <em>
+              {i18n.translate(
+                'xpack.graph.settings.advancedSettings.diversityFieldInputHelpText2',
+                {
+                  defaultMessage:
+                    'This must be a single-term field or searches will be rejected with an error',
+                }
+              )}
+            </em>
+          </>
+        }
+        label={i18n.translate('xpack.graph.settings.advancedSettings.diversityFieldInputLabel', {
+          defaultMessage: 'Diversity field',
+        })}
+      >
+        <EuiComboBox
+          fullWidth
+          placeholder={i18n.translate(
+            'xpack.graph.settings.advancedSettings.diversityFieldInputOptionLabel',
+            { defaultMessage: '[No diversification]' }
+          )}
+          singleSelection={{ asPlainText: true }}
+          options={allFields.map(field => ({ label: field.name, value: field }))}
+          selectedOptions={
+            advancedSettings.sampleDiversityField
+              ? [
+                  {
+                    label: advancedSettings.sampleDiversityField.name,
+                    value: advancedSettings.sampleDiversityField,
+                  },
+                ]
+              : []
+          }
+          onChange={choices => {
+            updateSetting(
+              'sampleDiversityField',
+              choices.length === 1 ? choices[0].value : undefined
+            );
+          }}
+        />
+      </EuiFormRow>
+
+      {advancedSettings.sampleDiversityField && (
+        <EuiFormRow
+          fullWidth
+          helpText={
+            <>
+              {i18n.translate('xpack.graph.settings.advancedSettings.maxValuesInputHelpText', {
+                defaultMessage:
+                  'Max number of documents in a sample that can contain the same value for the',
+              })}{' '}
+              <em>{advancedSettings.sampleDiversityField.name}</em>{' '}
+              {i18n.translate(
+                'xpack.graph.settings.advancedSettings.maxValuesInputHelpText.fieldText',
+                {
+                  defaultMessage: 'field',
+                }
+              )}
+            </>
+          }
+          label={i18n.translate('xpack.graph.settings.advancedSettings.maxValuesInputLabel', {
+            defaultMessage: 'Max docs per field',
+          })}
+        >
+          <EuiFieldNumber
```

Comment:
**Optional Enhancement**: Use an EuiRange with `showInput=true`


Although after posting all these enhancements, the max values all seem to be the same. Is this some arbitrary number or does it mean something? Number fields don't **require** a `max` value.

Reply:
Good question, I just copied those over from the existing form. I will ask and update accordingly.

Reply:
The max values from the old form were no hard upper limit, I removed them completely.